typescript/src/deposit-sweep.ts

```diff
@@ -130,238 +150,297 @@ export async function assembleDepositSweepTransaction(
+ * @param transaction - The transaction containing the input to be signed.
+ * @param inputIndex - Index pointing to the input within the transaction.
+ * @param deposit - Details of the deposit transaction.
+ * @param prevOutValue - The value from the previous transaction output.
+ * @param keyPair - A Signer object with the public and private key pair.
+ * @returns An empty promise upon successful signing.
  */
 async function signP2SHDepositInput(
-  transaction: any,
+  transaction: Transaction,
   inputIndex: number,
   deposit: Deposit,
-  walletKeyRing: any
-): Promise<void> {
-  const { walletPublicKey, depositScript, previousOutputValue } =
-    await prepareInputSignData(transaction, inputIndex, deposit, walletKeyRing)
+  prevOutValue: number,
+  keyPair: Signer
+) {
+  const { walletPublicKey, depositScript } = await prepareInputSignData(
+    deposit,
+    prevOutValue,
+    keyPair
+  )
+
+  const sigHashType = Transaction.SIGHASH_ALL
 
-  const signature: Buffer = transaction.signature(
+  const sigHash = transaction.hashForSignature(
     inputIndex,
     depositScript,
-    previousOutputValue,
-    walletKeyRing.privateKey,
-    bcoin.Script.hashType.ALL,
-    0 // legacy sighash version
+    sigHashType
   )
-  const scriptSig = new bcoin.Script()
-  scriptSig.clear()
-  scriptSig.pushData(signature)
-  scriptSig.pushData(Buffer.from(walletPublicKey, "hex"))
-  scriptSig.pushData(depositScript.toRaw())
-  scriptSig.compile()
-
-  transaction.inputs[inputIndex].script = scriptSig
+
+  const signature = script.signature.encode(keyPair.sign(sigHash), sigHashType)
+
+  const scriptSig: Stack = []
+  scriptSig.push(signature)
+  scriptSig.push(Buffer.from(walletPublicKey, "hex"))
+  scriptSig.push(depositScript)
+
+  transaction.ins[inputIndex].script = script.compile(scriptSig)
 }
 
 /**
- * Creates and sets witness script for the transaction input at the given index
- * by combining signature, wallet public key and deposit script.
- * @param transaction - Mutable transaction containing the input to be signed.
- * @param inputIndex - Index that points to the input to be signed.
- * @param deposit - Data of the deposit.
- * @param walletKeyRing - Key ring created using the wallet's private key.
- * @returns Empty promise.
+ * Signs a P2WSH deposit transaction input and sets the witness script.
+ * @param transaction - The transaction containing the input to be signed.
+ * @param inputIndex - Index pointing to the input within the transaction.
+ * @param deposit - Details of the deposit transaction.
+ * @param prevOutValue - The value from the previous transaction output.
+ * @param keyPair - A Signer object with the public and private key pair.
+ * @returns An empty promise upon successful signing.
  */
 async function signP2WSHDepositInput(
-  transaction: any,
+  transaction: Transaction,
   inputIndex: number,
   deposit: Deposit,
-  walletKeyRing: any
-): Promise<void> {
+  prevOutValue: number,
```

Comment:
`previousOutputValue`?

Reply:
Done in 98b622a5628e24d69b073ce22d713a8995ce0b2a.